float32 position
float32 max_speed
float32 acceleration
